zarf.yaml

```diff
@@ -39,6 +39,7 @@ components:
     actions:
       onDeploy:
         before:
+          - cmd: kubectl create configmap coredns-custom --namespace=kube-system --from-literal=uds.override='rewrite name regex (.*\.uds\.dev) host.k3d.internal' --dry-run=client -o yaml > core-dns-custom.yaml
```

Comment:
I don't have an answer that I'm happy with but I don't like writing out a file anywhere this deploys which could get accidentally committed (i.e. I run this in uds-core which doesn't `.gitignore` the dns yaml file).

A very simple solve would be `onDeploy.after` removing the local `core-dns-custom.yaml` file. But I almost lean towards the option of adding to the [dev stack chart](https://github.com/defenseunicorns/uds-k3d/tree/main/chart/templates) with an `onDeploy.after` to restart coredns? There's probably other options - open to whatever is least friction for end users.

Reply:
Yea, I had considered that.  Maybe it is a better option overall.  

I was also wondering if we'd need/want to customize the DNS override? Maybe other folks would want to specify similar rewrites.

Reply:
I'm not really sure what other rewrites someone might want to do - if they are using uds-k3d there are already assumptions about uds.dev being used as the domain (nginx routing to gateways, etc) so I don't think we need to consider other domains/allow that. 

Are there other situations you're thinking of here, or just thinking about the "maybe"? If it is "maybe", I don't think we need to optimize for "maybe" at this point - would prefer to keep this simple and handle requests if they arise.

Reply:
Also after thinking about this more would lean towards adding to the dev stack chart and doing a restart of coredns - that feels more transparent about this being "in the code" (and would be easier to read than the `kubectl create` in-line).

Reply:
Moved the config over to the uds-dev-stack and added a DNS.md docs